modules/sdk-coin-arbeth/test/unit/transferBuilder.ts

```diff
@@ -0,0 +1,187 @@
+
+import * as testData from '../resources';
+
+describe('Eth send multi sig builder', function () {
+  const toAddress = '0x7325A3F7d4f9E86AE62Cf742426078C3755730d5';
+  const xprv =
+    'xprv9s21ZrQH143K3D8TXfvAJgHVfTEeQNW5Ys9wZtnUZkqPzFzSjbEJrWC1vZ4GnXCvR7rQL2UFX3RSuYeU9MrERm1XBvACow7c36vnz5iYyj2';
+  const key = new KeyPair({ prv: xprv }).getKeys().prv as string;
+  const amount = '100000000000000000'; // equivalent to 0.1 ether
+
+  describe('should build', () => {
+    it('native coin transfer should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('tarbeth')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key)
+        .data('0x');
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_DATA);
+    });
+
+    it('native coin transfer with coin explicitly set should succeed', async () => {
+      const builder = new TransferBuilder()
+        .expirationTime(1590078260)
+        .coin('tarbeth')
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key)
+        .data('0x');
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_DATA);
+    });
+
+    it('native coin transfer with sequenceId zero should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('tarbeth')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(0)
+        .key(key)
+        .data('0x');
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_SEQUENCE_ZERO_DATA);
+    });
+
+    it('native coin transfer with amount 0 should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('tarbeth')
+        .expirationTime(1590078260)
+        .amount('0')
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key)
+        .data('0x');
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_AMOUNT_ZERO_DATA);
+    });
+
+    it('ERC20 token transfer should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('tarbeth:link')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_ARBETH_LINK_DATA);
+    });
+
+    it('erc20 transfer should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('tarbeth:link')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(0)
+        .key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_TOKEN_SEQUENCE_ZERO_DATA);
+    });
```

Comment:
What is special with zero sequence id so that we need to do an extra test for it?

Reply:
It seems like previously we didn't support sequence id from 0 and this change was introduced later, hence this testcase was added for eth, https://github.com/BitGo/BitGoJS/commit/bdc53412b61076967dd4b325b70fafb87a07dc6d